docs/design/batching.md

```diff
@@ -0,0 +1,64 @@
+# Batching VolumeSnapshotMover Backups and Restores
+
+## Summary
+The VolumeSnapshotMover controller currently runs on each `volumeSnapshotBackup` 
+and `volumeSnapshotRestore` until completed. The problem with this is that as 
+the number of VolSync processes increases, the time for backup/restore to complete
+can become slow, as diminishing returns increases.
+
+
+## Performance Testing Results
+[Performance testing](https://docs.google.com/document/d/1kPKo46_McEkj0Fu9hdkTxiVDnTh00forFQnscTgiEgg/edit?usp=sharing) 
+found that there is an optimal batch number of 12 for backup. For restore, 
+the results appear to be linear. 
+- It is important to note that these results are
+from a small set of factors such a CephFS and Ceph RBD, as well as small
+volume sizes. It is possible for this number to change with different scenarios.
+
+
+## Motivation
+Improve the performance of VolumeSnapshotMover backup and restore by allowing 
+users to specify a concurrent number of `volumeSnapshotBackups` and 
+`volumeSnapshotRestores` that can be simultaneously `inProgress` at once.
+
+
+## Implementation details
+Configurable *int64 values, `dpa.spec.dataMover.concurrentBackupVolumes` and 
+`dpa.spec.dataMover.concurrentRestoreVolumes`, can be used to specify 
+a number of `volumeSnapshotBackup` and `volumeSnapshotRestore` CRs that should 
+be operated on at once. For example, given 100 PVs, a user may want to use 
+VolumeSnapshotMover to backup 15 at a time and restore 20 at a time for
+improved performance. 
+
+To implement this in the the VSM controller, we can keep track of the current 
```

Comment:
@eemcmullan Can you please elaborate more on how are we gonna keep track of the VSBs/VSRs ? Will it be the case that we will be making List calls to the kube api server whenever the controller receives a reconcile event ? And then from this List we will look at the status of each VSB/VSR CRs  ? Won't this degrade the controller performance ? I may be missing something and be totally wrong here. Let me know your thoughts on this.

Reply:
@dymurray @shawn-hurley if possible, I would appreciate thoughts on this matter. Thanks in advance

Reply:
Rather than keeping track of a counter, I would do a lookup in the cache for all the ones that are in progress.

Reply:
+1, I think in previous discussions we weren't sure if the list call was cached, if it is I think that is absolutely the right move.

Reply:
makes sense, let go ahead with list call + queue approach